tests/helpers.py

```diff
@@ -153,9 +153,17 @@ def create_pod(
             pod_data['spec']['volumes'][0]['persistentVolumeClaim']['claimName'] = pvc_name
 
     if interface_type == constants.CEPHBLOCKPOOL and raw_block_pv:
-        pod_data['spec']['containers'][0]['volumeDevices'][0]['devicePath'] = raw_block_device
-        pod_data['spec']['containers'][0]['volumeDevices'][0]['name'] = pod_data.get('spec').get('volumes')[
-            0].get('name')
+        if pod_dict_path == constants.FEDORA_DC_YAML:
+            temp_dict = [
+                {'devicePath': raw_block_device, 'name': pod_data.get('spec').get(
```

Comment:
I guess this is the name of volume? can you rename this variable name accordingly?

Reply:
@keesturam which param ? raw_block_device ?
copied below content from logs, this is how it will be yaml.
`                                                                               volumeDevices:
                                                                               - devicePath: /dev/block
                                                                                 name: fedora-vol`

Reply:
@ramkiperiy /dev/block is already present in rwx pod pvc attach might fail if we use this path

Reply:
@prsurve as you can check in pastebin log, pod creation was success but looks like there is problem, I just checked df -kh data inside pod which is missing the block device. let me debug this further.
[rperiyas@localhost ocs-ci]$ oc rsh -n fiotest pod-test-rbd-288f76026f50458586aedf1bc3636086-1-qw2l5
sh-5.0# df -kh
Filesystem      Size  Used Avail Use% Mounted on
overlay         119G  9.7G  110G   9% /
tmpfs            64M     0   64M   0% /dev
tmpfs            32G     0   32G   0% /sys/fs/cgroup
shm              64M     0   64M   0% /dev/shm
tmpfs            32G  5.9M   32G   1% /etc/hostname
/dev/nvme0n1p3  119G  9.7G  110G   9% /etc/hosts
tmpfs            32G     0   32G   0% /proc/acpi
tmpfs            32G     0   32G   0% /proc/scsi
tmpfs            32G     0   32G   0% /sys/firmware
sh-5.0#

Reply:
I have seen similar issue with my PR in #1201 , I think /dev/block usage in yaml was error which we missed to notice, Also the other problem I see is once this fails, I cannot get a clean run again until I cleanup the cluster, not sure if its something to do with rook or something we are missing in yaml(the pvc shows its mounted but I dont see the mounted volume at all)

Reply:
@prsurve @vasukulkarni  Checked this and didn't see any problem by using /dev/block in DC pod, more details are below.
**1. Nginx pod, with /dev/block**
[rperiyas@localhost scripts]$ oc describe pod -n test pod-test-rbd-c92f069d8b4b48a1b0863f84e51a3986
Name:               pod-test-rbd-c92f069d8b4b48a1b0863f84e51a3986
Containers:
  my-container:
    Container ID:   cri-o://5532c98a1f9b5fd08df49782e4e89a74a9685a2fb961a25b2b33ab160d912111
    **Image:          nginx**
    Image ID:       docker.io/library/nginx@sha256:36b77d8bb27ffca25c7f6f53cadd059aca2747d46fb6ef34064e31727325784e
    Port:           <none>
    Host Port:      <none>
    State:          Running
      Started:      Wed, 08 Jan 2020 11:50:59 +0530
    Ready:          True
    Restart Count:  0
    Environment:    <none>
    Mounts:
      /var/run/secrets/kubernetes.io/serviceaccount from default-token-l4qgj (ro)
    **Devices:
      /dev/block from my-volume**
------------Inside nginx pod--------------------------
[rperiyas@localhost scripts]$ oc rsh -n test pod-test-rbd-c92f069d8b4b48a1b0863f84e51a3986
# ls /dev/block
/dev/block
# lsblk -rbno SIZE /dev/block
21474836480

**2. Fedora DC pod, with /dev/block**
[rperiyas@localhost scripts]$ oc describe pod -n test pod-test-rbd-ca9b5dc8836844ed963e5ef164c32b25-1-c2szj
Name:               pod-test-rbd-ca9b5dc8836844ed963e5ef164c32b25-1-c2szj
Containers:
  fedora:
    Container ID:  cri-o://09d9262a1d7bcd5f7b39b6580db37762c5db3adf7bd67eb533e1edc956cf7f15
    Image:         fedora
    Image ID:      docker.io/library/fedora@sha256:8fa60b88e2a7eac8460b9c0104b877f1aa0cea7fbc03c701b7e545dacccfb433
    Port:          <none>
    Host Port:     <none>
    Command:
      /bin/bash
      -ce
      tail -f /dev/null
    State:          Running
      Started:      Wed, 08 Jan 2020 11:51:50 +0530
    Ready:          True
    Restart Count:  0
    Limits:
      cpu:     150m
      memory:  500Mi
    Requests:
      cpu:        150m
      memory:     500Mi
    Liveness:     exec [sh -ec df /mnt] delay=3s timeout=1s period=3s #success=1 #failure=3
    Environment:  <none>
    Mounts:       <none>
    Devices:      <none>
      /dev/block from fedora-vol
------------------Inside fedora dc pod-------------------------
sh-5.0# ls /dev/block
/dev/block
sh-5.0#  lsblk -rbno SIZE /dev/block
21474836480

As you see above /dev/block is mapped in dc based pod and starting IO's also works without any issues. As you guys mentioned, a month before I did observed the problem of /dev/block not getting mapped to DC based pods, but it looks like working not sure what got changed.

Reply:
> @prsurve as you can check in pastebin log, pod creation was success but looks like there is problem. Just checked df -kh data inside pod, it's missing the block device. let me debug this further.

for above comment "df -kh" and "lsblk" --> checked in both nginx and DC based pod the behavior is same, looks like we can't list the mapped /dev/block using "df -kh" and "lsblk" commands. Using "ls /dev/block" cmd the mapped block is listing both nginx and dc pods.
**Command output from nginx**
[rperiyas@localhost scripts]$ oc rsh -n test pod-test-rbd-c92f069d8b4b48a1b0863f84e51a3986
# 
# lsblk
NAME                                                                                                  MAJ:MIN RM   SIZE RO TYPE MOUNTPOINT
loop0                                                                                                   7:0    0   340G  0 loop 
loop1                                                                                                   7:1    0    20G  0 loop 
loop2                                                                                                   7:2    0    20G  0 loop 
rbd0                                                                                                  252:0    0    40G  0 disk 
rbd1                                                                                                  252:16   0    20G  0 disk 
rbd2                                                                                                  252:32   0    20G  0 disk 
nvme0n1                                                                                               259:0    0   120G  0 disk 
|-nvme0n1p1                                                                                           259:1    0   384M  0 part 
|-nvme0n1p2                                                                                           259:2    0   127M  0 part 
|-nvme0n1p3                                                                                           259:3    0     1M  0 part 
`-nvme0n1p4                                                                                           259:4    0 119.5G  0 part 
  `-coreos-luks-root-nocrypt                                                                          253:0    0 119.5G  0 dm   /dev/termination-log
nvme1n1                                                                                               259:5    0    10G  0 disk 
nvme2n1                                                                                               259:6    0   340G  0 disk 
`-ceph--f12b4bee--1b6b--4c6c--876d--cd05c62ce672-osd--block--97dac58a--748f--40f0--8671--eee8980f7ea4 253:1    0   339G  0 lvm  
# df -kh
Filesystem                            Size  Used Avail Use% Mounted on
overlay                               120G   26G   94G  22% /
tmpfs                                  64M     0   64M   0% /dev
tmpfs                                  31G     0   31G   0% /sys/fs/cgroup
shm                                    64M     0   64M   0% /dev/shm
tmpfs                                  31G   13M   31G   1% /etc/hostname
/dev/mapper/coreos-luks-root-nocrypt  120G   26G   94G  22% /etc/hosts
tmpfs                                  31G   24K   31G   1% /run/secrets/kubernetes.io/serviceaccount
tmpfs                                  31G     0   31G   0% /proc/acpi
tmpfs                                  31G     0   31G   0% /proc/scsi
tmpfs                                  31G     0   31G   0% /sys/firmware
# ls /dev/block
/dev/block
# lsblk -rbno SIZE /dev/block
21474836480
#

**Command output from Fedora DC pod**
[rperiyas@localhost scripts]$ oc rsh -n test pod-test-rbd-ca9b5dc8836844ed963e5ef164c32b25-1-c2szj
sh-5.0# lsblk
NAME        MAJ:MIN RM   SIZE RO TYPE MOUNTPOINT
loop0         7:0    0   340G  0 loop 
loop1         7:1    0    20G  0 loop 
loop2         7:2    0    20G  0 loop 
rbd0        252:0    0    40G  0 disk 
rbd1        252:16   0    20G  0 disk 
rbd2        252:32   0    20G  0 disk 
nvme0n1     259:0    0   120G  0 disk 
|-nvme0n1p1 259:1    0   384M  0 part 
|-nvme0n1p2 259:2    0   127M  0 part 
|-nvme0n1p3 259:3    0     1M  0 part 
`-nvme0n1p4 259:4    0 119.5G  0 part 
nvme1n1     259:5    0    10G  0 disk 
nvme2n1     259:6    0   340G  0 disk 
sh-5.0# df -kh
Filesystem                            Size  Used Avail Use% Mounted on
overlay                               120G   26G   94G  22% /
tmpfs                                  64M     0   64M   0% /dev
tmpfs                                  31G     0   31G   0% /sys/fs/cgroup
shm                                    64M     0   64M   0% /dev/shm
tmpfs                                  31G   13M   31G   1% /etc/hostname
/dev/mapper/coreos-luks-root-nocrypt  120G   26G   94G  22% /etc/hosts
tmpfs                                  31G     0   31G   0% /proc/acpi
tmpfs                                  31G     0   31G   0% /proc/scsi
tmpfs                                  31G     0   31G   0% /sys/firmware
sh-5.0# ls /dev/block
/dev/block
sh-5.0# lsblk -rbno SIZE /dev/block
21474836480
sh-5.0#

Reply:
thanks for the details, I think we should get rid of /dev/block to avoid confusion with system collision, I am going to run some tests as well, if you change it here I will just pick up your changes.

Reply:
I can make the change from **/dev/block** to **/dev/rbdblock** in constants.py, apart from constants.py, ocs_ci/templates/app-pods/raw_block_pod.yaml has /dev/block device path, Let me change both of this.

Reply:
@vasukulkarni @prsurve updated the changes, please review and approve the changes.